File: app/backend/java/payment-service/src/main/java/com/srbms/paymentservice/dto/WalletRequestDTO.java
package com.srbms.paymentservice.dto;

import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import java.math.BigDecimal;

public class WalletRequestDTO {

    @NotNull(message = "User ID is required")
    private Integer userId;

    @NotNull(message = "Amount is required")
    @DecimalMin(value = "1.00", message = "Minimum top up amount is 1.00")
    private BigDecimal amount;

    private String paymentMethodToken;
    private String description;

    public WalletRequestDTO() {}

    public Integer getUserId() {
        return userId;
    }

    public void setUserId(Integer userId) {
        this.userId = userId;
    }

    public BigDecimal getAmount() {
        return amount;
    }

    public void setAmount(BigDecimal amount) {
        this.amount = amount;
    }

    public String getPaymentMethodToken() {
        return paymentMethodToken;
    }

    public void setPaymentMethodToken(String paymentMethodToken) {
        this.paymentMethodToken = paymentMethodToken;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: app/backend/.net/SmartRide.PaymentService/Models/DTOs/WalletRequestDTO.cs
using System.ComponentModel.DataAnnotations;

namespace SmartRide.PaymentService.Models.DTOs;

public class WalletRequestDTO
{
    [Required(ErrorMessage = "User ID is required")]
    public int UserId { get; set; }

    [Required(ErrorMessage = "Amount is required")]
    [Range(typeof(decimal), "1.00", "79228162514264337593543950335", ErrorMessage = "Minimum top up amount is 1.00")]
    public decimal Amount { get; set; }

    public string? PaymentMethodToken { get; set; }
    public string? Description { get; set; }
}
